sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -96,19 +92,20 @@ public static SamplingResult samplingResult(Decision decision, Attributes attrib
    *
    * <p>This is meant for use by custom {@link Sampler} implementations.
    *
-   * <p>Use {@link #samplingResult(Decision, Attributes)} if you need attributes.
+   * <p>Use {@link #samplingResult(Decision, Attributes, TraceState)} if you need attributes.
    *
    * @param decision The decision made on the span.
+   * @param traceState The {@link TraceState} to use.
    * @return A {@link SamplingResult} with empty attributes and the provided {@code decision}.
    */
-  public static SamplingResult emptySamplingResult(Decision decision) {
+  public static SamplingResult emptySamplingResult(Decision decision, TraceState traceState) {
     switch (decision) {
       case RECORD_AND_SAMPLE:
-        return EMPTY_RECORDED_AND_SAMPLED_SAMPLING_RESULT;
+        return SamplingResultImpl.createWithoutAttributes(Decision.RECORD_AND_SAMPLE, traceState);
```

Comment:
If we use null (or some static final instance of TraceState) to indicate "use parent tracestate", we could keep the constant objects, conserving allocations.

Reply:
yeah, but the spec describes the behavior for null, which is to invalidate the tracestate. :(

Reply:
OK, that was unintentional. But still, we could do `private static final TraceState USE_PARENT = TraceState.builder.build()`  and then check if the object identity of the contained tracestate `== USE_PARENT`.

Reply:
I think the spec doesn't say anything about a `null` tracestate here, only an empty one: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample

> * A `Tracestate` that will be associated with the `Span` through the new
>  `SpanContext`.
>  Note: If the sampler returns an empty `Tracestate` here, the `Tracestate` will be cleared,
>  so samplers should normally return the passed-in `Tracestate` if they do not intend
>  to change it.

Reply:
You're right. I read that wrong. But, I don't think using `null` as a signal to use the parent is a great API. I'd rather have an explicit way for Samplers to signal that, rather than an implicit null.

Reply:
Agreed, `null` isn't a great API here. So maybe the sentinel object I mentioned above?

Reply:
I'm not sure yet. My main concern is making sure that Sampler authors have a clear obvious way to do it. At the moment, the Sampler interface forces everyone to think about whether to modify the TraceState or not, and I think the most common use-case will be to not modify the TraceState, but just use the existing parent's. 

@Oberon00 do you think it's worth opening up a spec issue to make it easier to do the "right thing" in our SDK, or do you think the current spec is flexible enough to allow default-behavior to be implemented (using a sentinel value, or explicit boolean attribute)?

Reply:
I no one will complain that you violate the current spec if you make "use parent" the default somehow, but an editorial spec change / recommendation would make sense I think.

Reply:
I decided to start with an issue:  https://github.com/open-telemetry/opentelemetry-specification/issues/1031